### PR TITLE
Remove the local clones

### PR DESCRIPTION
**Pull request by zeripath**
In several places Gitea still makes full local clones instead of using a shared repository. 

This PR aims to remove these - using bare copies where it can.

### CODE REVIEW

**Comment by zeripath** [comment]
Unfortunately it's still the case that setting `SSH_ORIGINAL_COMMAND` causes the tests to fail as the hooks don't seem to work when running on the testcases.

**Comment by zeripath** [comment]
I've now enabled `SSH_ORIGINAL_COMMAND` wherever we internally push.

File: models/helper_directory.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	"github.com/Unknwon/com"
```
Comment (lunny) [inline models/helper_directory.go:17]: blank line

File: models/helper_directory.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// LocalCopyPath returns the local repository temporary copy path.
+func LocalCopyPath() string {
+	if filepath.IsAbs(setting.Repository.Local.LocalCopyPath) {
+		return setting.Repository.Local.LocalCopyPath
+	}
+	return path.Join(setting.AppDataPath, setting.Repository.Local.LocalCopyPath)
+}
+
+// CreateTemporaryPath creates a temporary path
+func CreateTemporaryPath(prefix string) (string, error) {
```
Comment (lunny) [inline models/helper_directory.go:29]: Is this compatible with existing temporary path and check out?

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/helper_directory.go (hunk 1 above)
Comment (zeripath) [reply models/helper_directory.go:17]: done

File: models/helper_directory.go (hunk 2 above)
Comment (zeripath) [reply models/helper_directory.go:29]: Yes. It's the same code but with a different prefix - I've moved the temp repo code to use it.

What really should be doing is removing temporary repo and moving that functionality to modules/git.

The main issue preventing that is the DiffIndex code.

**Comment by lunny** [comment]
Will fix #6544

File: models/helper_directory.go (hunk 2 above)
Comment (zeripath) [reply models/helper_directory.go:29]: Temp repo is now mostly a shim around these modules/git functions

**Comment by zeripath** [comment]
This severely conflicts with #6314 in ways that will require more thought. It also likely severely conflicts with @filipnavara 's #6478 . I'm therefore going to close this.

**Comment by filipnavara** [comment]
I think it deserves to be revisited but at the moment I already spent hours to resolve conflicts on my PR so I'm to tired to look into doing it here too.

**Comment by zeripath** [comment]
Aha! the issue is that #6314 can't cope with the hooks being run!

**Pull request closed without merging**